flink-runtime/src/test/java/org/apache/flink/runtime/minicluster/MiniClusterITCase.java

```diff
@@ -686,6 +708,42 @@ public void initializeOnMaster(ClassLoader loader) {
         }
     }
 
+    @Test
+    public void testCleanUpHaDataWhenClosingMiniCluster() throws Exception {
+        final int numOfTMs = 3;
+        final int slotsPerTM = 7;
+
+        File haDataFolder = temporaryFolder.newFolder();
+        String haDataFolderSuffix = "default";
+        File fullHaDataFolder = new File(haDataFolder, haDataFolderSuffix);
+
+        Configuration configuration = getDefaultConfiguration();
+        configuration.setString(HighAvailabilityOptions.HA_MODE, "ZOOKEEPER");
+        configuration.setString(HighAvailabilityOptions.HA_STORAGE_PATH, haDataFolder.toString());
+        configuration.setString(HighAvailabilityOptions.HA_ZOOKEEPER_QUORUM, zkServer.getConnectString());
+
+        final MiniClusterConfiguration cfg =
+                new MiniClusterConfiguration.Builder()
+                        .setNumTaskManagers(numOfTMs)
+                        .setNumSlotsPerTaskManager(slotsPerTM)
+                        .setRpcServiceSharing(RpcServiceSharing.DEDICATED)
+                        .setConfiguration(configuration)
+                        .build();
+
+        try (final MiniCluster miniCluster = new MiniCluster(cfg)) {
+            miniCluster.start(true);
+            // Check if ha data is still there
+            assertTrue(fullHaDataFolder.list().length > 0);
+
+            miniCluster.executeJobBlocking(getSimpleJob(numOfTMs * slotsPerTM));
+
+            miniCluster.close();
+
+            // check that ha data got removed
+            assertTrue(fullHaDataFolder.list().length == 0);
```

Comment:
Thanks for adding the test! However, we want to specifically test the inverse of this condition for the mini cluster enetrypoint, i.e. the HA data will not be removed.